core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -366,9 +366,12 @@ private[spark] class MesosClusterScheduler(
     } else {
       ""
     }
+
+    val sslOps = conf.getSSLConf.toArray.map({ case (k, v) => s"-D$k=$v" }).mkString(" ")
     val envBuilder = Environment.newBuilder()
     desc.command.environment.foreach { case (k, v) =>
-      envBuilder.addVariables(Variable.newBuilder().setName(k).setValue(v).build())
+      val vv = if (k == "SPARK_SUBMIT_OPTS") s"$v $sslOps" else v
```

Comment:
This is not in spark-submit itself, but in the dispatcher. SPARK_JAVA_OPTS might be a better place.